point_pattern.py

```diff
@@ -0,0 +1,92 @@
+
+
+from point import Point
+import analytics
+import utils
+import random
+import numpy as np
+
+class PointPattern(object):
+    def __init__(self):
+        self.points = []
+
+    def add_point(self, point):
+        self.points.append(point)
+
+    def remove_point(self, index):
+        try:
+            del(self.points[index])
+        except:
+            pass
+
+    def average_nearest_neighbor_distance(self, mark=None):
+        return analytics.average_nearest_neighbor_distance(self.points, mark)
+
+    def num_of_coincident(self):
+        numCoin = 0
+        accounted = []
+
+
+        for i in range(len(self.points)):
```

Comment:
Do you have a magic methods that could handle some of this for you?  Also, if A == B does B == A?